### PR TITLE
OBPIH-5823 Fulfilment alert to requestor - Approvals email alerts

### PR DESCRIPTION
**Pull request by alannadolny**
Some changes will probably be needed after @drodzewicz PR (this one with newStatus discussed during TH - will be done after merging the mentioned PR, etc.), but I am pushing it for review.

### CODE REVIEW

**Review by awalkowiak — CHANGES_REQUESTED** [review]
Please change wording from `fulfil...` into `fulfill...` and remove the `newStatus` param, other than that it LGTM

**Review by awalkowiak — APPROVED** [review]

File: grails-app/services/org/pih/warehouse/inventory/RequisitionStatusTransitionEventService.groovy [hunk 1]
```diff
@@ -25,5 +25,6 @@ class RequisitionStatusTransitionEventService implements ApplicationListener<Req
         Requisition requisition = event.requisition
         requisitionService.triggerRequisitionStatusTransition(requisition, event.createdBy)
         notificationService.publishRequisitionStatusTransitionNotifications(requisition)
+        notificationService.publishFulfillmentNotifications(requisition)
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/inventory/RequisitionStatusTransitionEventService.groovy:28]: This is just a state transition so it should probably go into the publishRequisitionStatusTransitionNotifications method when we have a chance to refactor.

**Review by jmiranda — COMMENTED** [review]
Minor refactoring requested.

File: grails-app/services/org/pih/warehouse/inventory/RequisitionStatusTransitionEventService.groovy (hunk 1 above)
Comment (drodzewicz) [reply grails-app/services/org/pih/warehouse/inventory/RequisitionStatusTransitionEventService.groovy:28]: I will move it in my open PR #4298

**Pull request merged**